packages/uui-popover-container/lib/uui-popover-container.element.ts

```diff
@@ -0,0 +1,285 @@
+
+    let top = 0;
+    let left = 0;
+
+    if (isBottomPlacement) {
+      top = targetRect.top + targetRect.height;
+      if (isStart) {
+        left = targetRect.left;
+      }
+      if (isEnd) {
+        left = targetRect.left + targetRect.width - popoverRect.width;
+      }
+      if (!isStart && !isEnd) {
+        left = targetRect.left + targetRect.width / 2 - popoverRect.width / 2;
+      }
+    }
+    if (isTopPlacement) {
+      top = targetRect.top - popoverRect.height;
+      if (isStart) {
+        left = targetRect.left;
+      }
+      if (isEnd) {
+        left = targetRect.left + targetRect.width - popoverRect.width;
+      }
+      if (!isStart && !isEnd) {
+        left = targetRect.left + targetRect.width / 2 - popoverRect.width / 2;
+      }
+    }
+    if (isLeftPlacement) {
+      left = targetRect.left - popoverRect.width;
+      if (isStart) {
+        top = targetRect.top;
+      }
+      if (isEnd) {
+        top = targetRect.top + targetRect.height - popoverRect.height;
+      }
+      if (!isStart && !isEnd) {
+        top = targetRect.top + targetRect.height / 2 - popoverRect.height / 2;
+      }
+    }
+    if (isRightPlacement) {
+      left = targetRect.left + targetRect.width;
+      if (isStart) {
+        top = targetRect.top;
+      }
+      if (isEnd) {
+        top = targetRect.top + targetRect.height - popoverRect.height;
+      }
+      if (!isStart && !isEnd) {
+        top = targetRect.top + targetRect.height / 2 - popoverRect.height / 2;
+      }
+    }
+
+    // Clamp left and top within screen bounds
+    // If the target leaves the screen, the popover follows.
+    const screenWidth = window.innerWidth;
+    const screenHeight = window.innerHeight;
+
+    const topTargetVsScreenTop = Math.min(
+      0,
+      targetRect.top + targetRect.height
+    );
+    const topTargetVsScreenBottom = Math.max(
+      Math.min(top, screenHeight - popoverRect.height),
+      targetRect.top - popoverRect.height
+    );
+
+    top = Math.max(topTargetVsScreenTop, topTargetVsScreenBottom);
+
+    const leftTargetVsScreenLeft = Math.min(
+      0,
+      targetRect.left + targetRect.width
+    );
+    const leftTargetVsScreenRight = Math.max(
+      Math.min(left, screenWidth - popoverRect.width),
+      targetRect.left - popoverRect.width
+    );
+
+    left = Math.max(leftTargetVsScreenLeft, leftTargetVsScreenRight);
+
+    // Detect if the popover is completely outside the screen on any side
+    const isCompletelyOutsideScreen =
+      top + popoverRect.height < 0 ||
+      top > screenHeight ||
+      left + popoverRect.width < 0 ||
+      left > screenWidth;
+
+    if (isCompletelyOutsideScreen) {
+      // @ts-ignore - This is part of the new popover API, but typescript doesn't recognize it yet.
+      this.hidePopover();
+    }
+
+    // Set the popover's position
+    this.style.transform = `translate(${left}px, ${top}px)`;
+  };
+
+  #findAncestorWithAttribute(
```

Comment:
This is the second time you have utilized this method.

Good, but actually there is no need for this to be part of the class and then we can avoid duplicating it.

So it can be extracted and moved to a utility file. then the sane method can be used in both places.